src/main/java/tech/ydb/io/r2dbc/state/OutTransaction.java

```diff
@@ -17,42 +17,76 @@
 package tech.ydb.io.r2dbc.state;
 
 import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
 
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
-import tech.ydb.io.r2dbc.result.YdbDMLResult;
-import tech.ydb.io.r2dbc.result.YdbDDLResult;
+import tech.ydb.core.UnexpectedResultException;
+import tech.ydb.io.r2dbc.query.OperationType;
+import tech.ydb.io.r2dbc.result.YdbResult;
 import tech.ydb.io.r2dbc.util.ResultExtractor;
 import tech.ydb.table.TableClient;
+import tech.ydb.table.query.DataQueryResult;
 import tech.ydb.table.query.Params;
 import tech.ydb.table.transaction.TxControl;
 
 /**
  * @author Kirill Kurdyukov
  */
-final class OutTransaction implements YdbConnectionState {
+public final class OutTransaction implements YdbConnectionState {
 
     private final TableClient tableClient;
     private final TxControl<?> txControl;
     private final Duration connectionTimeout;
 
-    OutTransaction(TableClient tableClient, TxControl<?> txControl, Duration connectionTimeout) {
+    public OutTransaction(TableClient tableClient, TxControl<?> txControl, Duration connectionTimeout) {
         this.tableClient = tableClient;
         this.txControl = txControl;
         this.connectionTimeout = connectionTimeout;
     }
 
     @Override
-    public Mono<YdbDMLResult> executeDataQuery(String yql, Params params) {
+    public Flux<YdbResult> executeDataQuery(String yql, Params params, List<OperationType> operationTypes) {
         return Mono.fromFuture(tableClient.createSession(connectionTimeout))
-                .map(sessionResult -> ResultExtractor.extract(sessionResult, "Error creating session"))
-                .flatMap(session -> Mono.fromFuture(session.executeDataQuery(yql, txControl, params)))
-                .map(dataQueryResultResult -> new YdbDMLResult(dataQueryResultResult.getValue()));
+                .flatMap(sessionResult -> ResultExtractor.extract(sessionResult, "Error creating session"))
+                .flatMapMany(session ->
```

Comment:
We can't make a smaller flatMap, because the result is Flux<YdbResult>, and we have Mono<Flux<...<YdbResult>> and we need to expand the result